runtime/parser2/statement.go

```diff
@@ -84,3 +91,106 @@ func parseReturnStatement(p *parser) *ast.ReturnStatement {
 		},
 	}
 }
+
+func parseIfStatement(p *parser) *ast.IfStatement {
+
+	var ifStatements []*ast.IfStatement
+
+	for {
+		startPos := p.current.Range.StartPos
+		p.next()
+
+		expression := parseExpression(p, lowestBindingPower)
+		if expression == nil {
+			panic(fmt.Errorf("expected test expression"))
+		}
+
+		thenBlock := parseBlock(p)
+		if thenBlock == nil {
+			panic(fmt.Errorf("expected block for then branch"))
+		}
+
+		var elseBlock *ast.Block
+
+		parseNested := false
+
+		p.skipSpaceAndComments(true)
+		if p.current.IsString(lexer.TokenIdentifier, "else") {
```

Comment:
Eventually would be nice to use a constant here, `KeywordIf` or something similar!

Reply:
Good point, I'll add constants for all the keywords 👍

Reply:
Added in 46bc216dfda6f3afa4e5ee036d1decb87a2dd10c